### PR TITLE
feature: support IP Masquerade agent

### PR DESCRIPTION
**Pull request by diegolnasc**

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@diegolnasc
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Review by bharathkkb — COMMENTED** [review]
Thanks for the PR @diegolnasc and for updating the tests!

Tests seem to be failing at `verify-simple-composer-env-v1` due to a diff reported by TF.

```
TestSimpleComposerEnvV1Module 2022-07-20T00:32:37Z cmd.go:107: Running terraform with args [plan -input=false -detailed-exitcode -lock=false]
TestSimpleComposerEnvV1Module 2022-07-20T00:32:37Z command.go:100: Running command terraform with args [plan -input=false -detailed-exitcode -lock=false]
TestSimpleComposerEnvV1Module 2022-07-20T00:32:40Z command.go:179: random_id.random_suffix: Refreshing state... [id=rcI]
TestSimpleComposerEnvV1Module 2022-07-20T00:32:40Z command.go:179: random_string.suffix: Refreshing state... [id=iyo3]
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179: google_compute_network.main: Refreshing state... [id=projects/PROJECT_ID/global/networks/ci-composer-test-iyo3]
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179: google_compute_subnetwork.main: Refreshing state... [id=projects/PROJECT_ID/regions/us-central1/subnetworks/ci-composer-test-iyo3]
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179: module.simple-composer.module.simple-composer-environment.google_composer_environment.composer_env: Refreshing state... [id=projects/PROJECT_ID/locations/us-central1/environments/composer-env-adc2]
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179: 
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179: Terraform used the selected providers to generate the following execution
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179: plan. Resource actions are indicated with the following symbols:
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179:   ~ update in-place
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179: 
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179: Terraform will perform the following actions:
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179: 
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179:   # google_compute_subnetwork.main will be updated in-place
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179:   ~ resource "google_compute_subnetwork" "main" {
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179:         id                         = "projects/PROJECT_ID/regions/us-central1/subnetworks/ci-composer-test-iyo3"
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179:         name                       = "ci-composer-test-iyo3"
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179:       - private_ip_google_access   = true -> null
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179:         # (11 unchanged attributes hidden)
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179:     }
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179: 
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179: Plan: 0 to add, 1 to change, 0 to destroy.
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179: 
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179: ─────────────────────────────────────────────────────────────────────────────
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179: 
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179: Note: You didn't use the -out option to save this plan, so Terraform can't
TestSimpleComposerEnvV1Module 2022-07-20T00:32:41Z command.go:179: guarantee to take exactly these actions if you run "terraform apply" now.
    terraform.go:296: 
        	Error Trace:	terraform.go:296
        	            				simple_composer_env_v1_test.go:30
        	            				terraform.go:327
        	            				terraform.go:347
        	            				stages.go:31
        	            				terraform.go:347
        	Error:      	Should not be: 2
        	Test:       	TestSimpleComposerEnvV1Module
        	Messages:   	plan after apply should have non-empty diff
```

**Comment by comment-bot-dev** [comment]
@diegolnasc
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: test/integration/go.mod [hunk 1]
```diff
@@ -0,0 +1,44 @@
+module github.com/terraform-google-modules/terraform-google-composer/test/integration
+
+go 1.17
+
+require (
+	github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test v0.0.0-20220401205256-9f9a444009fd
```
[the remark this answers is not in the document]
Comment (diegolnasc) [reply test/integration/go.mod]: Done.

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**